sklearn/tree/tests/test_tree.py

```diff
@@ -2155,7 +2173,13 @@ def test_decision_tree_regressor_sample_weight_consistentcy(criterion):
     # Thresholds, tree.tree_.threshold, and values, tree.tree_.value, are not
     # exactly the same, but on the training set, those differences do not
     # matter and thus predictions are the same.
-    assert_allclose(tree1.predict(X), tree2.predict(X))
+    y_pred1 = tree1.predict(X)
+    y_pred2 = tree2.predict(X)
+    # TODO: y_pred1 and y_pred2 will not match global_dtype for 32 bit
+    if global_dtype == np.float64:
+        assert y_pred1.dtype == global_dtype
+        assert y_pred2.dtype == global_dtype
```

Comment:
Note that solely dtype preservation should be asserted for fitted attributes when this is implemented. https://github.com/scikit-learn/scikit-learn/pull/22763/files aims at performing such assertion for `Regressors` generally.

Hence, I would remove this part:
```suggestion
```